cache/store.go

```diff
@@ -21,22 +21,27 @@ import (
 	"github.com/google/btree"
 
 	"go.etcd.io/etcd/api/v3/mvccpb"
+	"go.etcd.io/etcd/api/v3/v3rpc/rpctypes"
 	clientv3 "go.etcd.io/etcd/client/v3"
 )
 
 var ErrNotReady = fmt.Errorf("cache: store not ready")
 
+// The store keeps a bounded history of snapshots using ringBuffer so that
+// reads at historical revisions can be served until they fall out of the window.
 type store struct {
-	mu        sync.RWMutex
-	tree      *btree.BTree
-	degree    int
-	latestRev int64
+	mu              sync.RWMutex
+	liveTree        *btree.BTree
+	degree          int
+	latestRev       int64
+	snapshotHistory ringBuffer[*snapshot]
```

Comment:
```suggestion
	degree          int
	latest snapshot
	history ringBuffer[*snapshot]
```
Combination of tree and rev is what? A snapshot.